system/jlib/jptree.cpp

```diff
@@ -2981,6 +3014,156 @@ IPropertyTree *getXPathMatchTree(IPropertyTree &parent, const char *xpath)
     return matchTree;
 }
 
+void PTree::serializeAttributes(IBufferedSerialOutputStream &tgt) const
+{
+    Owned<IAttributeIterator> aIter = getAttributes();
+    if (aIter->first())
+    {
+        do
+        {
+            const char *name = aIter->queryName();
+            const char *value = aIter->queryValue();
+            size32_t nameLen = strlen(name);
+            size32_t valueLen = strlen(value);
+
+            tgt.put(sizeof(nameLen), &nameLen);
```

Comment:
I don't think this matches the existing serialiazation - I think that is /0 terminated.

Same elsewhere.  The suggested unit test should catch this.